sources/packages/backend/apps/api/src/services/student-appeal/student-appeal.service.ts

```diff
@@ -102,49 +106,111 @@ export class StudentAppealService extends RecordDataModelService<StudentAppeal>
+   * @param applicationId application ID related to the appeal, when applicable.
    */
-  async hasExistingAppeal(applicationId: number): Promise<boolean> {
-    const existingAppeal = await this.repo
-      .createQueryBuilder("appeal")
-      .select("1")
-      .innerJoin("appeal.appealRequests", "appealRequests")
-      .innerJoin("appeal.application", "application")
-      .where("application.id = :applicationId", { applicationId })
-      .andWhere("appealRequests.appealStatus = :pending", {
-        pending: StudentAppealStatus.Pending,
-      })
-      .limit(1)
-      .getRawOne();
+  private async createStudentAppealNotification(
+    studentId: number,
+    entityManager: EntityManager,
+    applicationId?: number,
+  ): Promise<void> {
+    let student: Student;
+    let application: Application;
+    const studentSelector: FindOptionsSelect<Student> = {
+      id: true,
+      user: {
+        id: true,
+        firstName: true,
+        lastName: true,
+        email: true,
+      },
+      birthDate: true,
+    };
+    if (applicationId) {
+      // Load student and application data, if applicationId is provided.
+      application = await entityManager.getRepository(Application).findOne({
+        select: {
+          id: true,
+          applicationNumber: true,
+          student: studentSelector,
+        },
+        relations: { student: { user: true } },
+        where: { id: applicationId },
+        loadEagerRelations: false,
+      });
+      student = application.student;
+    } else {
+      // Load only student data.
+      student = await entityManager.getRepository(Student).findOne({
+        select: studentSelector,
+        relations: { user: true },
+        where: { id: studentId },
+        loadEagerRelations: false,
+      });
+    }
+    const ministryNotification: StudentSubmittedChangeRequestNotification = {
+      givenNames: student.user.firstName,
+      lastName: student.user.lastName,
+      email: student.user.email,
+      birthDate: student.birthDate,
+      applicationNumber: application?.applicationNumber ?? "not applicable",
+    };
+    return this.notificationActionsService.saveStudentSubmittedChangeRequestNotification(
+      ministryNotification,
+      entityManager,
+    );
+  }
 
-    return !!existingAppeal;
+  /**
+   * Checks if a student appeal exists.
+   * @param studentId student ID related to the appeal.
+   * @param options query options.
+   * - `appealFormName` form name of the appeal request to be checked.
+   * - `isStudentOnlyAppeal` flag to indicate if the appeal must not be associated with an application.
+   * - `applicationId` application ID related to the appeal to be checked. Ignored if `isStudentOnlyAppeal` is true.
+   * - `appealStatus` status of the appeal request to be checked.
+   * @returns true if exists, false otherwise.
+   */
+  async hasAppeal(
+    studentId: number,
+    options?: {
+      appealFormName?: string;
```

Comment:
Would it be better to use the form name enum ?

Reply:
Maybe, but the enum is for all forms, while this method is used for a subset of the enum.
I would refrain from changing it.